src/app/components/Caption/index.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable jsx-a11y/aria-role */
+/** @jsxRuntime classic */
+/** @jsx  jsx  */
+import { jsx } from '@emotion/react';
+import { useContext } from 'react';
+import pathOr from 'ramda/src/pathOr';
+import { ServiceContext } from '#app/contexts/ServiceContext';
+import Inline from '#app/legacy/containers/InlineContainer';
+import InlineLink from '#app/legacy/containers/InlineLink';
+import Blocks from '#app/legacy/containers/Blocks';
+import { OptimoBlock } from '#app/models/types/optimo';
+import VisuallyHiddenText from '../VisuallyHiddenText';
+import styles from './index.style';
+import Fragment from '../Fragment';
+
+const componentsToRender = {
+  fragment: Fragment,
+  urlLink: InlineLink,
+  inline: Inline,
+};
+
+const chooseOffscreenText = (
+  mediaType: string,
+  videoCaption: string,
+  imageCaption: string,
+  audioCaption: string,
+  defaultText: string,
+) => {
+  switch (mediaType) {
+    case 'video':
+      return videoCaption;
+    case 'image':
+      return imageCaption;
+    case 'audio':
+      return audioCaption;
+    default:
+      return defaultText;
+  }
+};
+
+const renderParagraph = (block: OptimoBlock) => {
+  const paragraphBlock = pathOr(null, ['model', 'blocks'], block);
+
+  if (!paragraphBlock) return null;
+
+  return (
+    <p key={pathOr(null, ['0', 'id'], paragraphBlock)}>
+      <Blocks blocks={paragraphBlock} componentsToRender={componentsToRender} />
+    </p>
+  );
+};
+
+const renderCaption = (
+  paragraphBlocks: OptimoBlock[],
+  offscreenText: string,
+  dir: string,
+) => (
+  <figcaption
+    css={[
+      styles.captionStyles,
+      dir === 'rtl' ? styles.rtlStyles : styles.ltrStyles,
+    ]}
+    dir={dir}
+  >
```

Comment:
It may be worth trying to use our `<Text>` component. Its polymorphic, so you can use the `as='figcaption'` prop to render a `<figcaption>` and the use the `size` and `fontVariant` props to set the font styling, for example:
```
          <Text
            size="longPrimer"
            fontVariant="sansRegular"
            as="figcaption"
            css={styles.captionStyles}
          >
          ...
```

This means you could remove the manual font styling in the `captionStyles` class itself.